plugins/pipelineTemplates/create.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const urlLib = require('url');
+const boom = require('@hapi/boom');
+const schema = require('screwdriver-data-schema');
+const validator = require('screwdriver-template-validator').parsePipelineTemplate;
+const templateSchema = schema.api.templateValidator;
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/pipelineTemplates',
+    options: {
+        description: 'Create a new pipeline template',
+        notes: 'Create a specific pipeline template',
+        tags: ['api', 'pipelineTemplate'],
+        auth: {
+            strategies: ['token'],
+            scope: ['build']
+        },
+
+        handler: (request, h) => {
```

Comment:
```suggestion
        handler: (request, h) => {
                    const { pipelineTemplateVersionFactory, pipelineTemplateFactory } = request.server.app;

              const config = await validator(request.payload.yaml);
  
              if (config.errors.length > 0) {
                  throw boom.badRequest(`Template has invalid format: ${config.errors.length} error(s).`, config.errors);
              }
  
              const pipelineTemplate = await pipelineTemplateFactory.get({
                  name: config.template.name,
                  namespace: config.template.namespace
              });
              const { pipelineId } = request.auth.credentials;
  
              if (pipelineTemplate) {
                  if (pipelineTemplate.pipelineId !== pipelineId) {
                      throw boom.notFound(`Pipeline IDs do not match`);
                  }
              }
  
              const template = await pipelineTemplateVersionFactory.create(
                  {
                      ...config.template,
                      pipelineId
                  },
                  pipelineTemplateFactory
              );
  
              const urlObject = {
                  pathname: `${request.path}/${template.id}`
              };
              const location = new URL(
                  urlObject,
                  `${request.server.info.protocol}://${request.headers.host}:${request.headers.port}`
              ).toString();
  
              return h.response(template.toJson()).header('Location', location).code(201);
        }
```

Reply:
https://hapi.dev/api/?v=21.3.2#-requestheaders `request.headers.port` appears to be invalid and has been covered up by this urlLib.format function